development/gcp/workflows/secrets-leak-detector.yaml

```diff
@@ -1,69 +1,79 @@
 main:
   params: [ event ]
   steps:
-    - search_secrets_leaks_in_logs:
-        call: http.post
+    - check:
+        call: json.decode
         args:
-          url: ${scan-logs-for-secrets-url}
-          auth:
-            type: OIDC
-          headers:
-            Content-Type: "application/cloudevents+json"
-          body: $${event}
+          data: $${event.data.message.data}
         result: step_result
-    - check_if_leaks_found:
+    - check_if_prowjob_finished:
         switch:
-          - condition: $${step_result.body.type == "prowjob.logs.leaks.found"}
-            next: move_gcs_bucket
-        next: return_result
-    - move_gcs_bucket:
-        call: http.post
-        args:
-          url: ${move-gcs-bucket-url}
-          auth:
-            type: OIDC
-          headers:
-            Content-Type: "application/cloudevents+json"
-            X-Cloud-Trace-Context: $${map.get(step_result.headers, "X-Cloud-Trace-Context")}
-          body: $${step_result.body}
-        result: step_result
-    - search_github_issues:
-        call: http.post
-        args:
-          url: ${search-github-issue-url}
-          auth:
-            type: OIDC
-          headers:
-            Content-Type: "application/cloudevents+json"
-            X-Cloud-Trace-Context: $${map.get(step_result.headers, "X-Cloud-Trace-Context")}
-          body: $${step_result.body}
-        result: step_result
-    - check_if_issue_found:
-        switch:
-          - condition: $${step_result.body.type == "sap.tools.github.leakissue.notfound"}
-            next: create_github_issue
-        next: return_result
-    - create_github_issue:
-        call: http.post
-        args:
-          url: ${create-github-issue-url}
-          auth:
-            type: OIDC
-          headers:
-            Content-Type: "application/cloudevents+json"
-            X-Cloud-Trace-Context: $${map.get(step_result.headers, "X-Cloud-Trace-Context")}
-          body: $${step_result.body}
-        result: step_result
-    - send_slack_message:
-        call: http.post
-        args:
-          url: ${send-slack-message-url}/secret-leak-found
-          auth:
-            type: OIDC
-          headers:
-            Content-Type: "application/cloudevents+json"
-            X-Cloud-Trace-Context: $${map.get(step_result.headers, "X-Cloud-Trace-Context")}
-          body: $${step_result.body}
-        result: step_result
-    - return_result:
-        return: $${step_result}
+          - condition: $${step_result.status == "success" or step_result.status == "failure" or step_result.status == "aborted"}
```

Comment:
Is it possible to simplify this condition to sth like this? Would be easier to read and/or extend in future.

condition: $${step_result.status in ["success", "failure", "aborted"]}

Reply:
According to documentation it should work.